django/core/management/commands/shell.py

```diff
@@ -64,11 +65,13 @@ def ipython(self, options):
                 pass
             else:
                 return
-        # no IPython, raise ImportError
-        raise ImportError("No IPython")
+        raise NoSuchShell("No IPython")
```

Comment:
This still hides any `ImportError` if it's raised during the startup of IPython. I think you need to raise `NoSuchShell` when importing specific IPython versions fails, but before trying to initialize the shell, and let any other errors during initialization (including `ImportError`) bubble up.